data/provider/src/main/java/tech/pegasys/teku/api/ChainDataProvider.java

```diff
@@ -585,4 +585,20 @@ public void requireStoreAvailable() {
         .join()
         .map(Merkleizable::hash_tree_root);
   }
+
+  public SafeFuture<List<BlockHeader>> getBlockHeaders(
+      final Optional<Bytes32> parentRoot, final Optional<UInt64> slot) {
+    if (!isStoreAvailable()) {
+      throw new ChainDataUnavailableException();
+    }
+    if (parentRoot.isPresent()) {
+      return SafeFuture.completedFuture(List.of());
+    }
+
+    return combinedChainDataClient
+        .getBlockAtSlotExact(slot.orElse(recentChainData.getCurrentSlot().orElseThrow()))
```

Comment:
nit: Maybe extract this to a local `slotToLoad` variable as the chaining gets a bit confusing otherwise.  And maybe:
`slot.or(recentChainData::getCurrentSlot).orElseThrow()`?